File: TaskMasterWeb/TaskMasterWeb.Client/Dtos/User/UserDto.cs
namespace TaskMasterWeb.Client.Dtos.User;

public record UserDto
{
    public int Id { get; set; }

    public string Email { get; set; } = default!;

    public string FirstName { get; set; } = default!;

    public string LastName { get; set; } = default!;

    public string Role { get; set; } = default!;

    public IEnumerable<UserProjectDto>? Projects { get; set; }

    public IEnumerable<UserTaskDto>? Tasks { get; set; }
}

File: TaskMasterAPI/src/main/java/com/devlab/taskmaster/dto/user/UserDTO.java
package com.devlab.taskmaster.dto.user;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class UserDTO {

    private Integer id;

    private String email;

    private String firstName;

    private String lastName;

    private List<UserProjectDTO> projects;

    private List<UserTaskDTO> tasks;

    private String role;
}
